src/neuroconv/tools/nwb_helpers/_dataset_configuration.py

```diff
@@ -0,0 +1,195 @@
+    if isinstance(parent, NWBFile):
+        # Items in defined top-level places like acquisition, intervals, etc. do not act as 'containers'
+        # in the .parent sense; ask if object is in their in-memory dictionaries instead
+        for outer_field_name, outer_field_value in parent.fields.items():
+            if isinstance(outer_field_value, dict) and neurodata_object.name in outer_field_value:
+                return outer_field_name + "/" + neurodata_object.name + "/" + current_location
+        return neurodata_object.name + "/" + current_location
+    return _parse_location_in_memory_nwbfile(
+        current_location=neurodata_object.name + "/" + current_location, neurodata_object=parent
+    )
+
+
+def _get_dataset_metadata(
+    neurodata_object: Union[TimeSeries, DynamicTable], field_name: str, backend: Literal["hdf5", "zarr"]
+) -> Union[HDF5DatasetConfiguration, ZarrDatasetConfiguration]:
+    """Fill in the Dataset model with as many values as can be automatically detected or inferred."""
+    DatasetConfigurationClass = BACKEND_TO_DATASET_CONFIGURATION[backend]
+
+    candidate_dataset = getattr(neurodata_object, field_name)
+    # For now, skip over datasets already wrapped in DataIO
+    # Could maybe eventually support modifying chunks in place
+    # But setting buffer shape only possible if iterator was wrapped first
+    if not isinstance(candidate_dataset, DataIO):
+        # DataChunkIterator has best generic dtype inference, though logic is hard to peel out of it
+        # And it can fail in rare cases but not essential to our default configuration
+        try:
+            dtype = DataChunkIterator(candidate_dataset).dtype
+        except Exception as exception:
+            if str(exception) != "Data type could not be determined. Please specify dtype in DataChunkIterator init.":
+                raise exception
+            else:
+                dtype = np.dtype("object")
+
+        full_shape = get_data_shape(data=candidate_dataset)
+
+        if isinstance(candidate_dataset, GenericDataChunkIterator):
+            chunk_shape = candidate_dataset.chunk_shape
+            buffer_shape = candidate_dataset.buffer_shape
+        elif dtype != "unknown":
+            # TODO: eventually replace this with staticmethods on hdmf.data_utils.GenericDataChunkIterator
+            chunk_shape = SliceableDataChunkIterator.estimate_default_chunk_shape(
+                chunk_mb=10.0, maxshape=full_shape, dtype=np.dtype(dtype)
+            )
+            buffer_shape = SliceableDataChunkIterator.estimate_default_buffer_shape(
+                buffer_gb=0.5, chunk_shape=chunk_shape, maxshape=full_shape, dtype=np.dtype(dtype)
+            )
+        else:
+            pass  # TODO: think on this; perhaps zarr's standalone estimator?
+
+        location = _parse_location_in_memory_nwbfile(current_location=field_name, neurodata_object=neurodata_object)
+        dataset_name = location.strip("/")[-1]
```

Comment:
Oh, I think this was a suggestion you made at the retreat, and I propagated here but maybe it's still not removed from the higher up point?

The point to having it in the model was because you're not technically required to use this helper function to generate the model, is it OK to just keep it there and remove here (letting it be set by the model)?

Reply:
Removed, letting the DatasetInfo class handle it